core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2591,6 +2595,16 @@ private[spark] class CallerContext(
       }
     }
   }
+
+  def prepareContext(context: String): String = {
+    // The default max size of Hadoop caller context is 128
+    lazy val len = SparkHadoopUtil.get.conf.getInt("hadoop.caller.context.max.size", 128)
```

Comment:
I'm not sure we want to do this.  the 128 here is hardcoded and if hadoop changes its default then this is wrong.  It can change the default in the code without the config being present in the hadoop xml files.
  
while it would be nice to truncate to what hadoop actually is going to write I think its better to just send it and let hadoop do that.